Fixes the trash can not pulling up the owner notes

Burge337 opened the pull request:
This ended up being the same issue as the add note component; the x500 subscription completed after the program had already called the function that wanted to use it.

This also includes the add-note fix simply because it hadn't gotten merged into staging yet and it's hard to test a trash can if you don't have notes to delete.

Closes #35

Burge337 committed:
Fixes the trash can not pulling up the owner notes

This ended up being the same issue as the add note component; the x500 subscription completed after the program had already called the function that wanted to use it.

This also includes the add-note fix simply because it hadn't gotten merged into staging yet and it's hard to test a trash can if you don't have notes to delete.

Closes issue #35

helloworld12321 reviewed (changes requested):
Similarly to #33, this looks wonderful—looks good to merge as soon as there are Karma tests for `retreiveOwner()` :-)

helloworld12321 committed:
Merge branch 'staging' into fix-trash-in-staging

helloworld12321 committed:
Add Karma tests to TrashComponent

These tests all pass already.

As soon as the CI goes through, this branch should be ready to merge
into staging.

Addresses #29

helloworld12321 committed:
Merge remote-tracking branch 'refs/remotes/origin/staging'

Conflicts:
	client/src/testing/note.service.mock.ts

helloworld12321 committed:
Clean up after merge

(Mostly, making it so that we use the new note type rather than the old one.)

helloworld12321 committed:
Merge remote-tracking branch 'origin/staging' into fix-trash-in-staging

helloworld12321 committed:
Remove the floating action button on the Trash page

We used to use it for navigation, but now that functionality is handled by Ben B's awesome menu.

helloworld12321 committed:
Fix restoreNote in NoteController

(It still referred to "posted" instead of "status".)
